model-optimizer/extensions/front/tf/CTCGreedyDecoderReplacement.py

```diff
@@ -29,52 +35,115 @@ class CTCGreedyDecoderReplacement(FrontReplacementSubgraph):
-                                  'value': mask_value}).create_node()
-        decoder_node.in_port(1).disconnect()
-        decoder_node.in_port(1).connect(mask_node.out_port(0))
-
-        return {}
+        # obtain references to necessary nodes and their names
+        fill = match['fill']
+        dims = match['dims']
+        strided_slice = match['strided_slice']
+        strided_slice_1 = match['strided_slice_1']
+        ctc_greedy_decoder = match['decoder']
+        cast = match['cast']
+        sparse_to_dense = match['sparse_to_dense']
+        strided_slice_name = strided_slice.soft_get('name', strided_slice.id)
+        strided_slice_1_name = strided_slice_1.soft_get('name', strided_slice_1.id)
+        ctc_greedy_decoder_name = ctc_greedy_decoder.soft_get('name', ctc_greedy_decoder.id)
+        sparse_to_dense_name = sparse_to_dense.soft_get('name', sparse_to_dense.id)
+
+        # unsqueeze scalar values with batch size and time dimension
+        unsqueeze_batch_size = create_op_with_const_inputs(graph, Unsqueeze, {1: int64_array(0)},
+                                                           {'name': strided_slice_name + '/Unsqueeze'})
+        dims.in_port(0).get_connection().set_destination(unsqueeze_batch_size.in_port(0))
+        unsqueeze_time_size = create_op_with_const_inputs(graph, Unsqueeze, {1: int64_array(0)},
+                                                           {'name': strided_slice_1_name + '/Unsqueeze'})
+        fill.in_port(1).get_connection().set_destination(unsqueeze_time_size.in_port(0))
+
+        # compute a sequence mask shape [T, N] required for CTCGreedyDecoder
+        seq_mask_shape = Concat(graph, {'axis': 0, 'in_ports_count': 2,
+                                        'name': ctc_greedy_decoder_name + '/SequenceMaskShape'}).create_node()
+        seq_mask_shape.in_port(0).connect(unsqueeze_time_size.out_port(0))
+        seq_mask_shape.in_port(1).connect(unsqueeze_batch_size.out_port(0))
+
+        # compute a sequence mask
+        sequence_mask = create_op_with_const_inputs(graph, Broadcast, {0: np.array([1.0], dtype=np.float)},
+                                                    {'mode': 'numpy',
+                                                     'name': ctc_greedy_decoder_name + '/SequenceMask'})
+        sequence_mask.in_port(1).connect(seq_mask_shape.out_port(0))
+
+        # create CTCGreedyDecoder with the sequence mask instead of sequence length
+        ctc_greedy_decoder.in_port(1).disconnect()
+        ctc_greedy_decoder.in_port(1).connect(sequence_mask.out_port(0))
+
+        # remove fill and pack nodes since they are now in unconnected component
+        graph.remove_nodes_from([fill.id, dims.id])
+
+        # transform opset CTCGreedyDecoder output to TensorFlow's one that has a shape [N, T]
+        # opset CTCGreedyDecoder has an output with a shape [N, T, 1, 1]
+        squeeze_dec_seq = create_op_with_const_inputs(graph, Squeeze, {1: int64_array([2, 3])},
```

Comment:
This operation is a Squeeze from 4D to 2D and the MO will insert Transpose from NHWC to NCHW. Could you attach the screenshot of the part of the IR model with the sub-graph around the CTCGreedyDecoder? We may need to mark the Squeeze node with the `mark_input_as_in_correct_layout` function.

Reply:
![image](https://user-images.githubusercontent.com/35459624/91146171-3032d080-e6bf-11ea-83a3-b82ad62e0367.png)

Reply:
@lazarevevgeny , It seems no need to set this attribute. The screenshot is attached.